src/storage/src/storage_state/async_storage_worker.rs

```diff
@@ -100,28 +98,57 @@ where
         .await
         .expect("shard unavailable");
 
+    let remap_shard_write_handle_upper = write_handle.upper();
+
+    // Ensure that no collection's have advanced beyond the remap shard.
+    if let Some((id, resume_upper)) = resume_uppers
```

Comment:
This can happen and should be fine. If a source produces two collections of which one reaches the empty antichain but the other doesn't we want the reclocked collection for the first one to also reach the empty antichain but not the other.

So I think this check should be removed

Reply:
I forget why I thought this was necessary, but rewrote this code to just take the calculated resume upper and you're right that this is unnecessary.